modules/programs/retroarch.nix

```
@@ -0,0 +1,84 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.programs.retroarch;
+
+  enabledCores = lib.filterAttrs (_: core: core.enable) cfg.cores;
+  resolvedCores = lib.mapAttrsToList (
+    name: core:
+    if core.package != null then
+      core.package
+    else if builtins.hasAttr name pkgs.libretro then
+      builtins.getAttr name pkgs.libretro
+    else
+      throw "RetroArch core '${name}' not found in pkgs.libretro and no package specified"
+  ) enabledCores;
+in
+{
+  meta.maintainers = [
+    lib.hm.maintainers.jtrrll
+  ];
+
+  options.programs.retroarch = {
+    enable = lib.mkEnableOption "RetroArch";
+
+    package = lib.mkPackageOption pkgs "retroarch-bare" { };
+
+    cores = lib.mkOption {
+      type = lib.types.attrsOf (
+        lib.types.submodule (
+          { name, ... }:
+          {
+            options = {
+              enable = lib.mkEnableOption "RetroArch core";
+              package = lib.mkPackageOption pkgs.libretro name {
+                nullable = true;
+                default = null;
+                extraDescription = ''
+                  If null, will automatically use pkgs.libretro.''${name} if available.
+                '';
+              };
```

This is the correct way of handling this:

```suggestion
              package = lib.mkPackageOption pkgs "libretro.${name}" {
                default = [ "libretro" name ];
              };
```